File: ver-android/app/src/main/kotlin/org/opengamestudio/budget/budgetEffects.kt
package org.opengamestudio

import android.content.ClipboardManager
import android.content.ClipData
import android.content.Context

fun budgetCopyResult(
    ctx: Context,
    result: String
) {
    val clip = ctx.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    val item = ClipData.newPlainText("Result", result)
    clip.setPrimaryClip(item)
}

fun budgetPasteMorningBalance(ctx: Context) {
    val txt = budgetClipboardText(ctx) ?: ""
    budgetSet(F.pastedMorningBalance, txt)
}

fun budgetPasteSpent(ctx: Context) {
    val txt = budgetClipboardText(ctx) ?: ""
    budgetSet(F.pastedSpent, txt)
}

File: ver-ios/src/budget/budgetEffects.swift
import gb
import UIKit

func budgetCopyResult(_ result: String) {
    UIPasteboard.general.string = result
}

func budgetPasteMorningBalance() {
    let txt = UIPasteboard.general.string ?? "N/A"
    budgetSet(F.pastedMorningBalance, txt)
}

func budgetPasteSpent() {
    let txt = UIPasteboard.general.string ?? "N/A"
    budgetSet(F.pastedSpent, txt)
}
